core/src/main/java/com/sequenceiq/cloudbreak/service/credential/CredentialService.java

```diff
@@ -168,6 +168,14 @@ public Credential deleteByNameFromWorkspace(String name, Long workspaceId) {
         return delete(name, workspace);
     }
 
+    @Override
+    public Set<Credential> deleteMultipleByNameFromWorkspace(Set<String> names, Long workspaceId) {
+        Set<Credential> toBeDeleted = getByNamesForWorkspaceId(names, workspaceId);
```

Comment:
Why get the credentials just to get their names again? Is this to ensure that only credentials that are present are processed?

Reply:
Yeah, basically if you request a deletion of X, Y, and Z, and only X exists, we stop immediately and say "Hey, this isn't a valid request." So we actually wont' delete anything in this case.